google/cloud/forseti/scanner/audit/bigquery_rules_engine.py

```diff
@@ -147,17 +157,29 @@ def _build_rule(cls, rule_def, rule_index, raw_resource):
             raise audit_errors.InvalidRulesSchemaError(
                 'Faulty rule {}'.format(rule_def.get('name')))
 
-        rule_def_resource = bq_acls.BigqueryAccessControls(
-            project_id='',
-            dataset_id=escape_and_globify(dataset_id),
-            full_name='',
-            special_group=escape_and_globify(special_group),
-            user_email=escape_and_globify(user_email),
-            domain=escape_and_globify(domain),
-            group_email=escape_and_globify(group_email),
-            role=escape_and_globify(role.upper()),
-            view='',
-            raw_json=json.dumps(raw_resource))
+        def_mode = rule_def.get('mode')
+        if def_mode:
+            mode = Mode(def_mode)
+        else:
+            # Default mode to blacklist for backwards compatibility as that was
+            # the behaviour before mode was configurable.
+            # TODO: make mode required?
+            mode = Mode.BLACKLIST
+
+        rule_def_resource = RuleReference(
+            bigquery_acl=bq_acls.BigqueryAccessControls(
+                project_id='',
+                dataset_id=regular_exp.escape_and_globify(dataset_id),
+                full_name='',
+                special_group=regular_exp.escape_and_globify(special_group),
+                user_email=regular_exp.escape_and_globify(user_email),
+                domain=regular_exp.escape_and_globify(domain),
+                group_email=regular_exp.escape_and_globify(group_email),
+                role=regular_exp.escape_and_globify(role.upper()),
+                view={},
```

Comment:
Is there any reason why we updated the view from string to dictionary?

Reply:
I may have (incorrectly) seen it as a dict instead of a string somewhere and changed it. If it's supposed to be a string I can revert it.

Reply:
Found it: https://github.com/GoogleCloudPlatform/forseti-security/blob/c8d1485e5c88673d8c6190ba0e9a7c22013db385/google/cloud/forseti/common/gcp_type/bigquery_access_controls.py

The ACL class lists it as a dict which my linter caught. Either we should change the callers to use dict, or change the class to say the view is a string. I don't even know if the view is used, so it may be possible to delete it entirely...